File: src/java/foundgine-core/src/main/java/com/foundgine/core/semantic/metadata/ColumnMetadata.java
package com.foundgine.core.semantic.metadata;

import com.foundgine.core.abstractions.ColumnId;

public record ColumnMetadata(ColumnId id, String name, String storageName) {
	public ColumnMetadata(ColumnId id, String name) {
		this(id, name, null);
	}

	public String effectiveStorageName() {
		return storageName != null ? storageName : name;
	}
}

File: src/csharp/Foundgine.Core/Semantic/Metadata/ColumnMetadata.cs
using Foundgine.Core.Abstractions;

namespace Foundgine.Core.Semantic.Metadata;

public sealed record ColumnMetadata(
    ColumnId Id,
    string Name,
    string? StorageName = null)
{
    public string EffectiveStorageName => StorageName ?? Name;
}